orderer/common/cluster/clusterservice.go

```diff
@@ -163,7 +184,12 @@ func (s *ClusterService) VerifyAuthRequest(stream orderer.ClusterNodeService_Ste
 		return nil, errors.Errorf("node %d is not member of channel %s", authReq.ToId, authReq.Channel)
 	}
 
-	if !bytes.Equal(toIdentity, s.NodeIdentity) {
+	equal, err := compareCertPublicKeys(toIdentity, s.NodeIdentity)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to compare cert public keys")
+	}
```

Comment:
It seems there are not enough unit tests.

This is about the scenario.
1. The certificate is compared with the reference and everything is OK.
2. Something is changing.
3. And again it is compared with the standard.
bytes.Equal - should fail
compareCertPublicKeys - should be successful